statistics/handle/cache/statscacheinner.go

```diff
@@ -57,17 +59,21 @@ func WithTableStatsByQuery() TableStatsOpt {
 }
 
 // NewStatsCache creates a new StatsCacheWrapper.
-func NewStatsCache() *StatsCache {
+func NewStatsCache() (*StatsCache, error) {
 	enableQuota := config.GetGlobalConfig().Performance.EnableStatsCacheMemQuota
 	if enableQuota {
 		capacity := variable.StatsCacheMemQuota.Load()
-		return &StatsCache{
-			c: lru.NewStatsLruCache(capacity),
+		stats, err := lfu.NewLFU(capacity)
```

Comment:
In method `CopyAndUpdate`, the `StatsCache` copies the underlying `cache`, will that cause any risk since the LFU cache doesn't implement `copy` method?

Reply:
```
h.updateStatsCache(oldCache.CopyAndUpdate(tables, deletedTableIDs, opts...))
```

I want to merge ```updateStatsCache``` and ```CopyAndUpdate``` into a function. if users use ```LFU```. we do not use COW. But this change will be big. so I will do it in the next PR.

Reply:
It is at https://github.com/pingcap/tidb/pull/45518.

Reply:
Done